File: app/excel-search/src/getexcelfunctions/controller/searchAll/IndexController.java
package getexcelfunctions.controller.searchAll;

import getexcelfunctions.meta.ExcelFunctionsMeta;
import getexcelfunctions.model.ExcelFunctions;

import java.io.PrintWriter;
import java.util.List;

import org.slim3.controller.Controller;
import org.slim3.controller.Navigation;
import org.slim3.datastore.Datastore;

import com.google.appengine.api.datastore.Key;

public class IndexController extends Controller {

    @Override
    public Navigation run() throws Exception {
        String json;
        PrintWriter out = response.getWriter();
        List<Key> keys = Datastore.query().asKeyList();
        List<ExcelFunctions> datalist =
            Datastore.get(ExcelFunctions.class, keys);

        out.println("{");
        out.println("\"excelfunction\": [");
        int i;
        for (i = 0; i < datalist.size()-1; i++) {
            json = ExcelFunctionsMeta.get().modelToJson(datalist.get(i));
            out.println(json+",");
        }
        json = ExcelFunctionsMeta.get().modelToJson(datalist.get(i));
        out.println(json);
        out.println("]");
        out.println("}");

        return null;
    }
}

File: app/my-sheet/src/main/scala/com/aimluck/controller/aipoapps/sheet/IndexController.scala
package com.aimluck.controller.aipoapps.sheet
import org.slim3.controller.Navigation
import org.slim3.controller.Controller
import org.hidetake.opensocial.filter.model.OpenSocialRequest

class IndexController extends Controller {
  @throws(classOf[Exception])
  override def run(): Navigation = {
    val oRequest = OpenSocialRequest.create(request)
    response.getWriter().println("test")
    response.getWriter().println(oRequest.getViewerId())
    return null
  }

}